README_INTERFACE_TESTS.md

```diff
@@ -0,0 +1,283 @@
+# Interface tests
+
+Charm relation interfaces are information exchange protocols for Juju charms.
+In this repository, under each interface specification, it is possible to store a list of interface tests. The interface maintainers will write and maintain those tests for the interfaces they own. 
+The goal of the interface tests is to determine whether a given charm satisfies an interface protocol.
+
+There are two personas that need to work with interface tests: interface authors and interface users.
+If you are an interface user, i.e. the owner of a charm that uses some of the interfaces specified in this repository, then you should look at the documentation in the [interface-tester-pytest](https://github.com/canonical/interface-tester-pytest) repository.
+
+If you are an interface author, this page is for you.
+
+# How-To write interface tests as an interface author
+
+Throughout this example we will write some interface tests for the `ingress` interface.
+This is the expected behaviour for the interface:
+1) on `ingress-relation-joined`, the requirer should write its part of the data to the relation. 
+2) on `ingress-relation-changed`, if the provider has written its part of the data to the relation, then the provider should publish its part of the data; otherwise it should publish no data.
+
+## A minimal test case
+
+### Register a charm to the interface
+Write to `./interfaces/ingress/v0/charms.yaml`:
+
+```yaml
+providers:
+  - name: traefik-k8s
+    url: https://github.com/canonical/traefik-k8s-operator
+
+requirers: []
+```
+
+Verify that the `charms.yaml` file format is correct:
+
+> interface_tester discover --include ingress
```

Comment:
Use code blocks instead of blockquotes for terminal commands. This is applicable at a couple of places

Reply:
Fixed throughout